### PR TITLE
FIX `_get_items_to_delete` raising error when items list empty

### PR DESCRIPTION
**Pull request by Dr-Blank**
when calling `.reduce_size` on a newly created memory object without any cached items, it raises a `ValueError: min() arg is an empty sequence`. The argument of `age_limit` must not be None for this case.

Fixed by providing a default value to the `min` function used in calculating the oldest item, which would be the current datetime as no items exist.

### CODE REVIEW

**Review by tomMoral — COMMENTED** [review]
thanks a lot for the pr.!

maybe a simpler solution is to skip the function of no item is present?

also could you add a non regression test for this case?

File: joblib/_store_backends.py [hunk 1]
```diff
@@ -339,7 +339,10 @@ def _get_items_to_delete(
             to_delete_items = 0
 
         if age_limit is not None:
```
[the remark this answers is not in the document]
Comment (Dr-Blank) [reply joblib/_store_backends.py:341]: > maybe simpler:

Yes! much simpler.

On further thought, if the items are empty to begin with, why even perform other checks pertaining to size and item count, why not just return right away if no items are present?

```py
if not len(items) > 0:
    return []
```

**Comment by Dr-Blank** [comment]
Thank you for your comment! 🚀 

> also could you add a non regression test for this case?

I would certainly like to try it.

Full disclosure:
I am a beginner programmer and would love to attempt this, however at the time of reading I do not understand what regression tests mean 😅

**Comment by tomMoral** [comment]
A non-regression test is a test case in `test_memory.py`, that checks that calling `reduce_size` on an empty memory object and verify that it does not crash.

I agree that the simpler is to skip the function when the `items` list is empty.

**Review by tomMoral — APPROVED** [review]
LGTM! thanks for the PR @Dr-Blank !

**Comment by fcharras** [comment]
There's a test that failed on last merge on main related to reduce size and age limit on windows: https://dev.azure.com/joblib/joblib/_build/results?buildId=3725&view=logs&j=8008bb8c-6b4f-56d5-66f4-b1d037aba1bf&t=7af8808d-3494-56e0-c1dc-6fbf5bb2bc40&l=1563

Could it be related to those changes ?

**Comment by gothicVI** [comment]
Can we please get this sorted out and merged?

It's a pain running with the following construct:
```python
from datetime import timedelta
from joblib import Memory

cache_dir: Memory = Memory("/tmp/")  # nosec
try:
    cache_dir.reduce_size(age_limit=timedelta(days=1))
except ValueError:
    # https://github.com/joblib/joblib/pull/1503
    @cache_dir.cache(verbose=0)
    def dummy():
        """Define dummy function to initialize the cache."""

    dummy()
    cache_dir.reduce_size(age_limit=timedelta(days=1))

# define cached functions
```

**Pull request merged**